protocol_connect.go

```diff
@@ -903,25 +906,28 @@ func (m *connectUnaryRequestMarshaler) marshalWithGet(message any) *Error {
 	// TODO(jchadwick-buf): This function is mostly a superset of
 	// connectUnaryMarshaler.Marshal. This should be reconciled at some point.
 	if message == nil {
-		if m.writeWithGet(nil, false) {
-			return nil
-		}
-		return m.write(nil)
+		return m.writeWithGet(nil)
```

Comment:
I believe that this panics. Can we get some test coverage around this code path?